bpo-47000: Add `locale.getencoding()` (#32068)
python/cpython: Doc/library/locale.rst

   is not necessary or desired, *do_setlocale* should be set to ``False``.

   On Android or if the :ref:`Python UTF-8 Mode <utf8-mode>` is enabled, always
   return ``'utf-8'``, the :term:`locale encoding` and the *do_setlocale*
   argument are ignored.

   The :ref:`Python preinitialization <c-preinit>` configures the LC_CTYPE

>    locale. See also the :term:`filesystem encoding and error handler`.

Reviewer: I suggest to copy/paste this paragraph to getencoding() doc.

   .. versionchanged:: 3.7

>       The function now always returns ``"UTF-8"`` on Android or if the

Reviewer: ```suggestion
      The function now always returns ``"utf-8"`` on Android or if the
```

      :ref:`Python UTF-8 Mode <utf8-mode>` is enabled.

   .. versionchanged:: 3.11
      The function now returns ``"utf-8"`` instead of ``"UTF-8"`` on Android

>       or if the :ref:`Python UTF-8 Mode <utf8-mode>` is enabled.

Reviewer: I am sure that the upper case versus lower case is worth to be mentioned. But I'm fine with keeping if you consider that it's worth it.

   locale. See also the :term:`filesystem encoding and error handler`.

   .. versionchanged:: 3.7
      The function now always returns ``"UTF-8"`` on Android or if the
      :ref:`Python UTF-8 Mode <utf8-mode>` is enabled.

.. function:: getencoding()

>    Returns the :term:`locale encoding`.

Reviewer: I suggest this documentation:

```
Get the current :term:`locale encoding`:

* On Android and VxWorks, return ``"UTF-8"``.
* On Unix, return the encoding of the current :data:`LC_CTYPE` locale.
  Return ``"UTF-8"`` if :ref:`nl_langinfo(CODESET) <nl_langinfo>`
  returns an empty string: for example, if the current LC_CTYPE locale is not supported.
* On Windows, return the ANSI code page.
```

   locale. See also the :term:`filesystem encoding and error handler`.

   .. versionchanged:: 3.7
      The function now always returns ``"UTF-8"`` on Android or if the
      :ref:`Python UTF-8 Mode <utf8-mode>` is enabled.

.. function:: getencoding()

   Get the current :term:`locale encoding`:

   * On Android and VxWorks, return ``"UTF-8"``.
   * On Unix, return the encoding of the current :data:`LC_CTYPE` locale.
     Return ``"UTF-8"`` if ``nl_langinfo(CODESET)`` returns an empty string:
     for example, if the current LC_CTYPE locale is not supported.
   * On Windows, return the ANSI code page.

>    This function is same to ``getpreferredencoding(False)`` except this

Reviewer: ```suggestion
   This function is similar to :func:`getpreferredencoding(False) <getpreferredencoding>` except this
```

>    function ignore the :ref:`UTF-8 Mode <utf8-mode>`.

Reviewer: ```suggestion
   function ignores the :ref:`Python UTF-8 Mode <utf8-mode>`.
```
